search-endpoint.md

```diff
@@ -0,0 +1,397 @@
+# Discovery Search Endpoints
+[Back to the list of all defined endpoints](endpoints.md)
+
+## Search Endpoint
+**/api/discover/search**   
+
+This endpoint provides the functionality of the Discovery search screen: http://demo.dspace.org/xmlui/discover.
+
+Provide access to the Discovery search system (SOLR based). This endpoint returns the list of available DSpace object (DSO) types that can be searched:
+* all: Execute a query over all available DSO types.
+* items: Only search the DSpace items.
+* communities: Search within the DSpace community records.
+* collections: Limit the search to DSpace collection records.
+
+Example: TODO
+
+### Search with a given DSO type
+**/api/discover/search/<:dso-type>**
+
+This endpoint will provide detailed information on the Discovery search configuration that can be used to build complex searches.
+
+It supports the following parameters:
+* `scope`: UUID of a specific DSpace container (site, community or collection) to which the search will to be limited. If the scope has a specific DSpace configuration defined in `config/spring/api/discovery.xml`[https://github.com/DSpace/DSpace/blob/master/dspace/config/spring/api/discovery.xml#L28] that configuration will be returned. Otherwise the default configuration will be returned.
+
```

Comment:
This would require a small modification to the dspace-api so I guess we could do this.